### ISSUE TITLE
About the inference results of depth and point cloud.

### ISSUE THREAD

**Issue by shangshang98** [ByteDance-Seed/Depth-Anything-3#59]
Hello, thanks for your work！
I encountered an issue while running the code. Following your Basic Usage , when inferring with the image you provided, the depth map in the inference results (preditions) looks great, but at the same time, the gaussian.means (point cloud) is not properly formed. What could be the reason for this，thanks for your help.
The visualization results of the depth map and the point cloud are as follows:

<img width="520" height="294" alt="Image" src="https://github.com/user-attachments/assets/6fa7492b-a5bc-4c16-bd5b-0f6b4d019feb" />

<img width="486" height="314" alt="Image" src="https://github.com/user-attachments/assets/867654a9-d67c-4c63-9ddf-457387ff26e5" />

**Comment by pengcanon** [comment]
It looks right to me. fiddle/move/rotate around your pcd viewer until you find the right perspective. The pcd is derived from only two views which means the range of perspectives might be limited.

**Comment by shangshang98** [comment]
I used the "depth-anything/DA3NESTED-GIANT-LARGE" model. The code is as follows. The saved point cloud doesn't look quite right, with many scattered points on the sides, and I can't find a suitable angle to make it look normal.

````
import glob, os, torch
from depth_anything_3.api import DepthAnything3
import numpy as np
from einops import rearrange
import trimesh
device = torch.device("cuda")

model = DepthAnything3.from_pretrained("depth-anything/DA3NESTED-GIANT-LARGE") 
model = model.to(device=device)
example_path = "assets/examples/SOH"  
images = sorted(glob.glob(os.path.join(example_path, "*.png")))
prediction = model.inference(
    images,infer_gs=True
)
depthshow = (prediction.depth[0]).astype(np.uint8)
gaussians_means = rearrange(prediction.gaussians.means, "b (v n) c -> (b v n) c", v=2)
image = prediction.processed_images
image = rearrange(image, "v h w c -> (v h w) c")
all_ply_pred = trimesh.points.PointCloud(vertices=gaussians_means.detach().cpu().numpy(),
                                colors=(image.astype(np.uint8)))
all_ply_pred.export("all_pred.ply")
````

The results of "all_pred.ply" is as follows,
<img width="1215" height="474" alt="Image" src="https://github.com/user-attachments/assets/285aabac-d63b-4917-8be9-5e351f4ed27a" />
The results on the huggingface is as follows,

![Image](https://github.com/user-attachments/assets/9f5d41b0-bb7b-49de-bc8c-84d3e1b0a7dc)

**Comment by Jaykumaran** [comment]
You can Clone the huggingface spaces repo locally run and debug their script

https://huggingface.co/spaces/depth-anything/depth-anything-3/tree/main

**Comment by pengcanon** [comment]
> I used the "depth-anything/DA3NESTED-GIANT-LARGE" model. The code is as follows. The saved point cloud doesn't look quite right, with many scattered points on the sides, and I can't find a suitable angle to make it look normal.
> 
> ```
> import glob, os, torch
> from depth_anything_3.api import DepthAnything3
> import numpy as np
> from einops import rearrange
> import trimesh
> device = torch.device("cuda")
> 
> model = DepthAnything3.from_pretrained("depth-anything/DA3NESTED-GIANT-LARGE") 
> model = model.to(device=device)
> example_path = "assets/examples/SOH"  
> images = sorted(glob.glob(os.path.join(example_path, "*.png")))
> prediction = model.inference(
>     images,infer_gs=True
> )
> depthshow = (prediction.depth[0]).astype(np.uint8)
> gaussians_means = rearrange(prediction.gaussians.means, "b (v n) c -> (b v n) c", v=2)
> image = prediction.processed_images
> image = rearrange(image, "v h w c -> (v h w) c")
> all_ply_pred = trimesh.points.PointCloud(vertices=gaussians_means.detach().cpu().numpy(),
>                                 colors=(image.astype(np.uint8)))
> all_ply_pred.export("all_pred.ply")
> ```
> 
> The results of "all_pred.ply" is as follows, <img alt="Image" width="1215" height="474" src="https://private-user-images.githubusercontent.com/71620023/516046941-285aabac-d63b-4917-8be9-5e351f4ed27a.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjM1ODI5MDcsIm5iZiI6MTc2MzU4MjYwNywicGF0aCI6Ii83MTYyMDAyMy81MTYwNDY5NDEtMjg1YWFiYWMtZDYzYi00OTE3LThiZTktNWUzNTFmNGVkMjdhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTE5VDIwMDMyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA5OTk4MzI3NGE1YTg2ZGY5ZGYxNDNlMzI0NDY0OTE4YzM3MTlkMTdhNGNhNDkwNDk2YTAxMmI2ZjUyYmMxZDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.C8cAcRI7MjZTmvKG7c2iePCD1COvY3Vw9lOqII3O0zg"> The results on the huggingface is as follows,
> 
> ![Image](https://github.com/user-attachments/assets/9f5d41b0-bb7b-49de-bc8c-84d3e1b0a7dc)

Your pcd honestly seems correct to me, as I've seen this myself. Zoom in to those smaller chuncks and rotate around until you see the right spot. Some of the noisy dots you see in your viewer likely have low confidence values attached to them therefore will not show up in the hunggingface space because there, a threshold is set to filter out the noise points.

**Comment by daodao316** [comment]
Do you Fix it?
